raiden-ts/src/transport/epics.ts

```diff
@@ -83,6 +95,94 @@ import { getPresences$, getRoom$, roomMatch, globalRoomNames } from './utils';
 const AVAILABLE = ['online', 'unavailable'];
 const userRe = /^@(0x[0-9a-f]{40})[.:]/i;
 
+/**
+ * Joins the global broadcast rooms and returns the room ids.
+ *
+ * @param config - The {@link RaidenConfig} provides the broadcast room aliases for pfs and discovery.
+ * @param server - The matrix server used e.g., https://transport04.raiden.network is part of the alias.
+ * @param matrix - The {@link MatrixClient} instance used to create the filter.
+ * @returns Observable of the list of room ids for the the broadcast rooms.
+ */
+function joinGlobalRooms(
+  config: RaidenConfig,
+  server: string,
+  matrix: MatrixClient,
+): Observable<string[]> {
+  const serverName = getServerName(server);
+  return from(globalRoomNames(config)).pipe(
+    map(globalRoom => `#${globalRoom}:${serverName}`),
+    mergeMap(alias => matrix.joinRoom(alias)),
+    map(room => room.roomId),
+    toArray(), // wait all joinRoom promises to complete
+  );
+}
+
+/**
+ * Creates and returns a matrix filter. The filter reduces the size of the initial sync by
+ * filtering out broadcast rooms, emphemeral messages like receipts etc.
+ *
+ * @param matrix - The {@link MatrixClient} instance used to create the filter.
+ * @param roomIds - The ids of the rooms to filter out during sync.
+ * @returns Observable of the {@link Filter} that was created.
+ */
+function createFilter(matrix: MatrixClient, roomIds: string[]): Observable<Filter> {
+  return of(roomIds).pipe(
+    map(filteredRooms => ({
+      not_rooms: filteredRooms,
+      ephemeral: {
+        not_types: ['m.receipt'],
+      },
+      state: {
+        lazy_load_members: true,
+      },
+    })),
+    map((roomFilter: RoomFilterJson) => ({
+      room: roomFilter,
+      presence: {
+        types: ['m.presence'],
+      },
+    })),
+    mergeMap((filterDefinition: FilterDefinition) => from(matrix.createFilter(filterDefinition))),
+  );
+}
+
+function startMatrixSync(
+  state$: Observable<RaidenState>,
+  matrix$: AsyncSubject<MatrixClient>,
+  matrix: MatrixClient,
+  config$: Observable<RaidenConfig>,
+  server: string,
+) {
+  return state$
+    .pipe(
+      pluck('transport', 'matrix', 'server'),
+      filter((_server): _server is string => !!_server),
+      take(1),
+      tap(() => {
+        matrix$.next(matrix);
+        matrix$.complete();
+      }),
+      switchMapTo(matrix$),
+      delay(1e3), // wait 1s before starting matrix, so event listeners can be registered
+    )
+    .pipe(
+      withLatestFrom(config$),
+      mergeMap(([matrix, config]) =>
+        joinGlobalRooms(config, server, matrix).pipe(
+          mergeMap(roomIds => createFilter(matrix, roomIds)),
+          map(filter => [matrix, filter] as const),
+        ),
+      ),
+      mergeMap(([matrix, filter]) =>
+        matrix.startClient({
```

Comment:
This could go above, inside `joinGlobalRooms`'s pipe, replacing the `map as const` there. so you don't need to pass `matrix` down